modules/yieldmoBidAdapter.js

```diff
@@ -175,169 +229,223 @@ function getPageDescription() {
+ * @param {BidRequest} bidRequest bidder request object.
+ * @return Object OpenRTB's 'imp' (impression) object
  */
-function isDfp() {
-  try {
-    const frameElement = window.frameElement;
-    const parentElement = window.frameElement.parentNode;
-    if (frameElement && parentElement) {
-      return (
-        frameElement.id.indexOf('google_ads_iframe') > -1 &&
-        parentElement.id.indexOf('google_ads_iframe') > -1
-      );
+function openRtbImpression(bidRequest) {
+  const videoReq = utils.deepAccess(bidRequest, 'mediaTypes.video');
+  const size = extractPlayerSize(bidRequest);
+  const imp = {
+    id: bidRequest.bidId,
+    tagid: bidRequest.adUnitCode,
+    bidfloor: bidRequest.params.bidfloor || 0,
+    ext: {
+      placement_id: bidRequest.params.placementId
+    },
+    video: {
+      w: size[0],
+      h: size[1],
+      mimes: videoReq.mimes,
+      linearity: 1
     }
-    return false;
-  } catch (e) {
-    return false;
+  };
+
+  const videoParams = utils.deepAccess(bidRequest, 'params.video');
+  Object.keys(videoParams)
+    .filter(param => includes(OPENRTB_VIDEO_BIDPARAMS, param))
+    .forEach(param => imp.video[param] = videoParams[param]);
+
+  if (videoParams.skippable) {
+    imp.video.skip = 1;
   }
+
+  return imp;
 }
 
 /**
- * @returns true if there is an AMP context object
+ * @param {BidRequest} bidRequest bidder request object.
+ * @return [number, number] || null Player's width and height, or undefined otherwise.
  */
-function isAmp() {
-  try {
-    const ampContext = window.context || window.parent.context;
-    if (ampContext && ampContext.pageViewId) {
-      return ampContext;
-    }
-    return false;
-  } catch (e) {
-    return false;
+function extractPlayerSize(bidRequest) {
+  const sizeArr = utils.deepAccess(bidRequest, 'mediaTypes.video.playerSize');
+  if (utils.isArrayOfNums(sizeArr, 2)) {
+    return sizeArr;
+  } else if (utils.isArray(sizeArr) && utils.isArrayOfNums(sizeArr[0], 2)) {
+    return sizeArr[0];
   }
+  return null;
 }
 
 /**
- * @returns true if the environment is a SafeFrame.
+ * @param {BidRequest} bidRequest bid request object
+ * @param {BidderRequest} bidderRequest bidder request object
+ * @return Object OpenRTB's 'site' object
  */
-function isSafeFrame() {
-  return window.$sf && window.$sf.ext;
-}
+function openRtbSite(bidRequest, bidderRequest) {
+  let result = {};
 
-/**
- * @returns true if the environment is a dfp safe frame.
- */
-function isDFPSafeFrame() {
-  if (window.location && window.location.href) {
-    const href = window.location.href;
-    return (
-      isSafeFrame() &&
-      href.indexOf('google') !== -1 &&
-      href.indexOf('safeframe') !== -1
-    );
+  const loc = utils.parseUrl(utils.deepAccess(bidderRequest, 'refererInfo.referer'));
+  if (!utils.isEmpty(loc)) {
+    // result.page = `${loc.protocol}://${loc.hostname}${loc.pathname}`;  // commented because otherwise it doesn't work for QA rig page
```

Comment:
The line is commented because otherwise, it won't work for QA rig page. So, this is just for testing.